crates/sparrow-session/src/execution.rs

```diff
@@ -70,6 +78,16 @@ impl Execution {
         }
     }
 
+    /// Send the stop signal.
+    ///
+    /// This method does *not* wait for all batches to be processed.
```

Comment:
What do you mean by this? We will still process all in-flight batches, until each downstream component of the `Scan` receives the `None` message, right?

Reply:
Exactly -- it dosen't wait for those batches to be finished. Additionally, in-flight batches may still make their way out of the system after the call to stop returns.